packages/scripts/src/create-entrypoint.ts

```diff
@@ -386,13 +404,47 @@ function importStaticConfigs() {
     }`;
 }
 
+function addConfigsEventListenerForParentEnvironments(publicConfigsRoute: string) {
+    return `if(isMainEntrypoint) {
+        const fetchedConfigs = {};
+        const configsEventListener = async ({ data: { id, envName }, source }) => {
+            if(source && id === '${publicConfigsRoute}') {
```

Comment:
having so much untyped entry code in a string is problematic, and may cause maintenance burden.

the initial goal/approach for the entry generation code was to try and have it mostly typed. the generated entrypoint would ideally only pass build values to imported function calls. those functions (like `runEngineApp`) were themselves typed.

I suggest separating the parts that can be typed to ts files and importing their output in the entry code.

Reply:
I do agree. We already have the same notion in the engine-runtime-node package.

I think it can be a good first issue @AdamYahid

Reply:
created an issue https://github.com/wixplosives/engine/issues/911